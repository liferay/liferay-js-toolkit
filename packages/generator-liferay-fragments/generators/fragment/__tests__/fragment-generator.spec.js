const fs = require('fs');
const path = require('path');
const YeomanTest = require('yeoman-test');

function expectFile(...paths) {
  return expect(
    fs.readFileSync(path.join(...paths), 'utf-8')
  ).toMatchSnapshot();
}

function expectFiles(base, paths) {
  return paths.map(_path => expectFile(path.join(base, _path)));
}

describe('fragment-generator', () => {
  it('generates a new fragment', () =>
    YeomanTest.run(path.join(__dirname, '..'))
      .withOptions({ fragmentName: 'Sample Fragment' })
      .withOptions({ fragmentDescription: 'Sample Description' })
      .withOptions({ fragmentType: 'section' })
      .withOptions({ fragmentCollectionSlug: 'sample-collection' })
      .then(projectPath => {
        expectFiles(
          path.join(projectPath, 'src', 'sample-collection', 'sample-fragment'),
          ['fragment.json', 'index.html', 'styles.css', 'main.js']
        );
      }));

  it('accepts no description', () =>
    YeomanTest.run(path.join(__dirname, '..'))
      .withOptions({ fragmentName: 'Sample Fragment' })
      .withOptions({ fragmentType: 'section' })
      .withOptions({ fragmentCollectionSlug: 'sample-collection' })
      .then(projectPath => {
        expectFiles(
          path.join(projectPath, 'src', 'sample-collection', 'sample-fragment'),
          ['fragment.json', 'index.html', 'styles.css', 'main.js']
        );
      }));

  it('needs a name', () =>
    new Promise((resolve, reject) =>
      YeomanTest.run(path.join(__dirname, '..'))
        .withOptions({ fragmentDescription: 'Sample Description' })
        .withOptions({ fragmentType: 'section' })
        .withOptions({ fragmentCollectionSlug: 'sample-collection' })
        .then(reject)
        .catch(resolve)
    ));

  it('needs a collection', () =>
    new Promise((resolve, reject) =>
      YeomanTest.run(path.join(__dirname, '..'))
        .withOptions({ fragmentName: 'Sample Fragment' })
        .withOptions({ fragmentType: 'section' })
        .withOptions({ fragmentDescription: 'Sample Description' })
        .then(reject)
        .catch(resolve)
    ));
});
