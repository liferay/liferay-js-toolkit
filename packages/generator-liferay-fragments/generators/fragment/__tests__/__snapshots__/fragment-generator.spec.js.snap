// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragment-generator accepts no description 1`] = `
"{
  \\"cssPath\\": \\"styles.css\\",
  \\"htmlPath\\": \\"index.html\\",
  \\"jsPath\\": \\"main.js\\",
  \\"name\\": \\"Sample Fragment\\"
}
"
`;

exports[`fragment-generator accepts no description 2`] = `
"<div class=\\"sample-fragment\\">
  Sample Fragment
</div>
"
`;

exports[`fragment-generator accepts no description 3`] = `
".sample-fragment {
  color: inherit;
}
"
`;

exports[`fragment-generator accepts no description 4`] = `
"const content = fragmentElement.querySelector('.sample-fragment');
"
`;

exports[`fragment-generator generates a new fragment 1`] = `
"{
  \\"cssPath\\": \\"styles.css\\",
  \\"htmlPath\\": \\"index.html\\",
  \\"jsPath\\": \\"main.js\\",
  \\"name\\": \\"Sample Fragment\\"
}
"
`;

exports[`fragment-generator generates a new fragment 2`] = `
"<div class=\\"sample-fragment\\">
  Sample Fragment
</div>
"
`;

exports[`fragment-generator generates a new fragment 3`] = `
".sample-fragment {
  color: inherit;
}
"
`;

exports[`fragment-generator generates a new fragment 4`] = `
"const content = fragmentElement.querySelector('.sample-fragment');
"
`;
